Tella/Scenes/Dropbox/ViewModel/DropboxOutboxViewModel.swift

```diff
@@ -0,0 +1,217 @@
+                        self?.updateReportStatus(reportStatus: .submissionError)
+                        self?.toastMessage = error.errorMessage
+                        self?.shouldShowToast = true
+                    }
+                }
+            }, receiveValue: { [weak self] progressInfo in
+                self?.updateProgressInfos(uploadProgressInfo: progressInfo)
+            })
+            .store(in: &cancellables)
+    }
+    
+    override func pauseSubmission() {
+        if isSubmissionInProgress {
+            dropboxRepository.pauseUpload()
+            updateReportStatus(reportStatus: .submissionPaused)
+            
+            cancellables.removeAll()
+        }
+    }
+    
+    override func updateReportStatus(reportStatus: ReportStatus) {
+        self.reportViewModel.status = reportStatus
+        self.objectWillChange.send()
+        
+        guard let id = reportViewModel.id else { return }
+        
+        mainAppModel.tellaData?.updateDropboxReportStatus(reportId: id, status: reportStatus)
+    }
+    
+    override func updateCurrentFile(uploadProgressInfo: UploadProgressInfo) {
+        guard let dropboxProgressInfo = uploadProgressInfo as? DropboxUploadProgressInfo else {
+            return
+        }
+        
+        // Update the file's bytesSent and status
+        self.reportViewModel.files = self.reportViewModel.files.compactMap { file in
+            guard file.id == dropboxProgressInfo.fileId else { return file }
+            let updatedFile = file
+            updatedFile.bytesSent = dropboxProgressInfo.bytesSent ?? 0
+            updatedFile.status = dropboxProgressInfo.status
+            return updatedFile
+        }
+        
+        // Update the file in the database with offset and sessionId
+        if let file = self.reportViewModel.files.first(where: { $0.id == dropboxProgressInfo.fileId }) {
+            let dropboxFile = file
+            
+            dropboxFile.sessionId = dropboxProgressInfo.sessionId
+            dropboxFile.offset = dropboxProgressInfo.offset
+            
+            self.updateFile(file: dropboxFile)
+        }
+    }
+    
+    private func updateReportFolderId(folderId: String, name: String) {
+        guard let id = reportViewModel.id else { return }
+        
+        mainAppModel.tellaData?.updateDropboxFolderId(reportId: id, folderId: folderId,folderName: name )
+    }
+    
+    override func updateFile(file: ReportVaultFile) {
+        guard let dropboxFile = DropboxReportFile(reportFile: file) else { return }
+        
+        mainAppModel.tellaData?.updateDropboxReportFile(file: dropboxFile)
+        
+    }
+    
+    func reAuthenticateConnection() {
+        Task {
+            do {
+                try await dropboxRepository.handleSignIn()
+            } catch {
+                debugLog(error)
+            }
+        }
+    }
+    
+    func handleURLRedirect(url: URL) {
+        _ = dropboxRepository.handleRedirectURL(url) { authResult in
```

Comment:
maybe we need to handle success and failure ? like in `DropboxServerViewModel` class

    func handleURLRedirect(url: URL) {
        _ = dropboxRepository.handleRedirectURL(url) { [weak self] authResult in
            guard let self = self else { return }
            
            switch authResult {
            case .success:
                self.signInState = .loaded(true)
                self.addServer()
            case .error(_, let description):
                self.signInState = .error(description ?? "")
            default:
                break
            }
        }
    }